### PR TITLE
feat: Add cosign registry opts for provenance registry

### PR DESCRIPTION
**Pull request by saisatishkarra**
triggered on specification of COSIGN_REPOSITORY env

### CODE REVIEW

**Comment by laurentsimon** [comment]
Thanks. Let me know if you're OK with my latest comment https://github.com/slsa-framework/slsa-github-generator/issues/3024#issuecomment-1875722666

**Comment by saisatishkarra** [comment]
> Thanks. Let me know if you're OK with my latest comment [slsa-framework/slsa-github-generator#3024 (comment)](https://github.com/slsa-framework/slsa-github-generator/issues/3024#issuecomment-1875722666)

Sounds good to me!! lets get this merged and tested out against e2e in the main!! That should support the feature in the first place. As a second iteration we could an explicit variable in the verifier along side the environment variable

File: README.md [hunk 1]
```diff
@@ -274,6 +274,29 @@ Get the digest for your container _without_ pulling it using the [crane](https:/
 IMAGE="${IMAGE}@"$(crane digest "${IMAGE}")
 ```
 
+**Optional:**
```
Comment (laurentsimon) [inline README.md:277]: Let's remove this since it's going to be removed int he next PR when we introduce the `--provenance-registry`

**Review by laurentsimon — APPROVED** [review]
Thanks. let's just revert the README changes, since it will be obsolete in the next PR

**Review by laurentsimon — APPROVED** [review]

**Pull request merged**